packages/meditrak-server/src/routes/GETMapOverlays/assertMapOverlaysPermissions.js

```diff
@@ -2,72 +2,64 @@
  * Tupaia
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
-import { flattenDeep, keyBy } from 'lodash';
-import { hasAccessToEntityForVisualisation } from '../utilities';
-
+import { QUERY_CONJUNCTIONS } from '@tupaia/database';
 import { hasBESAdminAccess } from '../../permissions';
+import { hasAccessToEntityForVisualisation } from '../utilities';
 
-export const filterMapOverlaysByPermissions = async (accessPolicy, models, mapOverlays) => {
-  if (hasBESAdminAccess(accessPolicy)) {
-    return mapOverlays;
-  }
-
-  const allEntityCodes = [...new Set(flattenDeep(mapOverlays.map(m => m.countryCodes)))];
-  const entities = await models.entity.find({ code: allEntityCodes });
-  const entityByCode = keyBy(entities, 'code');
-  const accessCache = {}; //Cache the access so that we dont have to recheck for some map overlays
-  const filteredMapOverlays = [];
-
-  for (let i = 0; i < mapOverlays.length; i++) {
-    const mapOverlay = mapOverlays[i];
-    const countryCodes = mapOverlay.countryCodes || [];
-    const mapOverlayEntities = countryCodes.map(c => entityByCode[c]);
-    const permissionGroup = mapOverlay.userGroup;
-    const countryCodesString = countryCodes.join(',');
-
-    //permissionGroup and countryCodes are the 2 values for checking access for a map overlay
-    const cacheKey = `${permissionGroup}/${countryCodesString}`;
-    let hasAccessToMapOverlay = accessCache[cacheKey];
-
-    //Perform access checking if its not in the cache, otherwise reuse the value
-    if (hasAccessToMapOverlay === undefined) {
-      for (let j = 0; j < mapOverlayEntities.length; j++) {
-        const mapOverlayEntity = mapOverlayEntities[j];
-        hasAccessToMapOverlay = await hasAccessToEntityForVisualisation(
-          accessPolicy,
-          models,
-          mapOverlayEntity,
-          permissionGroup,
-        );
+const { RAW } = QUERY_CONJUNCTIONS;
 
-        //If users have access to any countries of the overlay,
-        //it should be enough to allow access the overlay
-        if (hasAccessToMapOverlay) {
-          break;
-        }
-      }
+export const assertMapOverlaysPermissions = async (accessPolicy, models, mapOverlayId) => {
+  const mapOverlay = await models.mapOverlay.findById(mapOverlayId);
 
-      accessCache[cacheKey] = hasAccessToMapOverlay;
-    }
+  const countryCodes = mapOverlay.countryCodes || [];
```

Comment:
Man `countryCodes` is a poorly named column now huh...could you add a card for renaming to reflect the fact that it also contains project codes?

Reply:
"now" being since the introduction of project level map overlays - nothing to do with your work.

Reply:
https://github.com/beyondessential/tupaia-backlog/issues/1489